File: backend/internal/model/transaction.go
package model

import (
	"time"

	"gorm.io/gorm"
)

// Transaction represents a financial transaction.
type Transaction struct {
	gorm.Model
	WalletID    uint 
	Wallet      Wallet `gorm:"constraint:OnUpdate:CASCADE;OnDelete:CASCADE"`
	CategoryID  uint
	Amount      float64
	Category    Category `gorm:"constraint:OnUpdate:CASCADE;OnDelete:RESTRICT"`
	Description string
	Date        time.Time
	IdempotencyKey *string `gorm:"size:64;uniqueIndex"`

	// Foreign keys
}

// TableName specifies the table name for Transaction model.
func (Transaction) TableName() string {
	return "transactions"
}

File: android/app/src/main/java/id/my/rizalanggoro/arta/domain/Transaction.kt
package id.my.rizalanggoro.arta.domain

import kotlinx.serialization.Serializable

@Serializable
data class Transaction(
    val id: Int,
    val walletId: Int,
    val amount: Double,
    val categoryId: Int,
    val category: Category? = null,
    val description: String = "",
    val date: String,
    val createdAt: String = "",
    val updatedAt: String = "",
)
